### PR TITLE
JBPM-6243 - Move deleteServerInstance to SpecManagementService

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

File: kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java [hunk 1]
```diff
@@ -420,6 +424,24 @@ public synchronized void stopContainer(ContainerSpecKey containerSpecKey) {
         notificationService.notify(serverTemplate, containerSpec, containers);
     }
 
+    @Override
+    public void deleteServerInstance(final ServerInstanceKey serverInstanceKey) {
+        checkNotNull("serverInstanceKey",
+                     serverInstanceKey);
+        System.out.println("getKieServerInstanceManager().isAlive(serverInstanceKey) = " + getKieServerInstanceManager().isAlive(serverInstanceKey));
```
Comment (mswiderski) [inline kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java]: let's remove system out from here

File: kie-server-parent/kie-server-controller/kie-server-controller-client/src/main/java/org/kie/server/controller/client/rest/RestKieServerControllerClient.java [hunk 2]
```diff
@@ -203,6 +203,11 @@ public void upgradeContainer(ContainerSpecKey containerSpecKey,
         throwUnsupportedException();
     }
 
+    @Override
+    public void deleteServerInstance(ServerInstanceKey serverInstanceKey) {
+        throwUnsupportedException();
```
Comment (sutaakar) [inline kie-server-parent/kie-server-controller/kie-server-controller-client/src/main/java/org/kie/server/controller/client/rest/RestKieServerControllerClient.java]: @cristianonicolai Do you plan to add support of this feature (and other currently unsupported features) for REST controller later?

File: kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java (hunk 1 above)
Comment (cristianonicolai) [reply kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java]: definitely :)

File: kie-server-parent/kie-server-controller/kie-server-controller-client/src/main/java/org/kie/server/controller/client/rest/RestKieServerControllerClient.java (hunk 2 above)
Comment (cristianonicolai) [reply kie-server-parent/kie-server-controller/kie-server-controller-client/src/main/java/org/kie/server/controller/client/rest/RestKieServerControllerClient.java]: @sutaakar I'm currently focusing on the websocket protocol as the primary option to be used from the workbench when connecting to the standalone controller. Once that is done I can look into extending the REST operations.

Comment (sutaakar) [reply kie-server-parent/kie-server-controller/kie-server-controller-client/src/main/java/org/kie/server/controller/client/rest/RestKieServerControllerClient.java]: Ok, thanks.

**Pull request merged**